src/dash-table/reconcile/number.ts

```diff
@@ -0,0 +1,33 @@
+import { INumberColumn } from 'dash-table/components/Table/props';
+import isNully from './isNully';
+
+export function coerce(value: any, options?: INumberColumn) {
+    const validation = validate(value, options);
+    if (validation.success) {
+        return validation;
+    }
+
+    switch (typeof value) {
+        case 'string':
+            return { success: Boolean(/(-)?\d+([.]\d+)?/.test(value)), value: parseFloat(value) };
```

Comment:
Depending on how complete you want to be here, there can be a lot of other numbers... https://github.com/plotly/fast-isnumeric

Reply:
Yeah, I did this to get basic support but switching to `fast-isnumeric` is trivial. Will update.

Reply:
Note that, though it's not really described there, `fast-isnumeric` is based on using type casting (plotly.js for example does [`+value`](https://github.com/plotly/plotly.js/blob/39915526327c657055feebc5e13b762616d52ebd/src/lib/coerce.js#L90) rather than `parseFloat`). It looks like the only case where `parseFloat` behaves differently (among values that pass `fast-isnumeric`) is hex strings (`'0xff'` - `parseFloat` just drops everything after the first number it finds), and it would be worthwhile deciding whether we want to accept hex (I would argue yes, and let it be converted to decimal unless and until we support hex formatting).

The other weird thing `parseFloat` does is drop any trailing junk (hence its handling of hex) which I don't think we want. Your test here, as written, ignores leading AND trailing junk... we definitely don't want that!

Reply:
I was young and optimistic?

Reply:
Supporting hex input sounds like a win to me. If we need hex presentation later (or type!), cool. I don't really see a scenario where someone should care that the value came in as `120` or as `0x78`...